### DISCUSSION TITLE
Unable to log in when using external authentication

### DISCUSSION THREAD

**Discussion by bjmd** [victorbalssa/abacus#267] [Bug]
**Describe the bug**
When trying to log in, it does not redirect me to the Oauth login page (it does if I try to log in with Oauth instead or PAT, but then it doesn't log in correctly). I then get an error message saying:

> Failed to validate accessToken, please re-check your token or backend URL. Error: Cannot read property of 'email' at undefined. 

I was previously not using external auth and I thought maybe I needed to force it to re-login, so I created a new PAT and tried to add credential but still throws the error. Also, I tried to delete the app so that it would start from scratch, but it remembers the old config when reinstalling and I can't see any local files so that's odd. 

![Screenshot 2024-03-23 at 19 51 12](https://github.com/victorbalssa/abacus/assets/9675756/493acd42-19b6-4eb2-9927-ec32f148539d)

**To Reproduce**
Type my backend URL into the app, along with a personal access token.

**Expected behavior**
It should forward to the Google auth page and then sign in with the PAT.

**Smartphone (please complete the following information):**

Device: iPhone 15
OS: iOS 17
Firefly III v6.1.12 running in Docker behind Traefik and traefik-forward-auth using google Oauth
Latest Abacus for Firefly III

**Comment by victorbalssa** [comment]
PAT won't work with `traefik-forward-auth`, you can try to whitelist `^/api/*`?

If you can't access the /api/* endpoints directly, PAT won't work.

**Reply by bjmd** [reply]
Thanks, I thought that may have been the case. 

Is there any way to get it working with the oauth authentication, or only by whitelisting /api?

**Reply by victorbalssa** [reply]
Unfortunately not with a forward to Google auth
